Public/Src/Engine/Dll/Distribution/RemoteWorker.cs

```diff
@@ -648,7 +670,10 @@ public async Task SendToRemote(OperationContext operationContext, RunnablePip ru
             }
             catch (InvalidOperationException)
             {
-                if (m_isConnectionLost)
+                // If the connection is lost or the worker is not running at all since the beginning 
+                // (due to ValidateCacheConnection failure), then fail the pips.
+                // Otherwise, it is an unexpected exception, so throw it.
+                if (m_isConnectionLost || !m_everAvailable)

```

Comment:
>m_everAvailable [](start = 43, length = 15)

shouldn't it be `EverAvailable` (i.e. Volatile.Read()) ?

Reply:
good catch!

---
In reply to: [348763766](https://github.com/microsoft/BuildXL/pull/1216#discussion_r348763766) [](ancestors = 348763766)